pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -191,6 +193,20 @@ private void start(List<Integer> indexList) {
 
     @Override
     CompletableFuture<MessageId> internalSendWithTxnAsync(Message<?> message, Transaction txn) {
+        if (txn != null && ((TransactionImpl)txn).getState() != TransactionImpl.State.OPEN) {
```

Comment:
Can we reuse the code from newly added in ProducerImpl?

Reply:
@liangyepianzhou LGTM, please check the above comment to avoid the duplicated code.